### PR TITLE
CART-947 fix: Treat all HG errors the same way

### PR DESCRIPTION
**Pull request by frostedcmos**
- Treat all HG errors from HG_Forward() the same way. Mercury
can now return number of different errors. All errors should be
treated identically, by being reported within the completion callback
instead of being returned directly to the caller.

- Change increment of posted handles to 0, preserving for now 2.0.0rc1
and prior behavior of mercury when exhausting number of pre-posted
handles.

Signed-off-by: Alexander Oganezov <alexander.a.oganezov@intel.com>

### CODE REVIEW

**Comment by frostedcmos** [comment]
I am awaiting verification on monday that this actually fixes all problems seen in CART-947, but even if not, those changes need to be landed for things to work properly going forward.

File: debian/control [hunk 1]
```diff
@@ -10,7 +10,7 @@ Build-Depends: debhelper (>= 10),
                libopenmpi-dev,
                libssl-dev,
                libyaml-dev,
-               libmercury-dev (= 2.0.0~rc1-1),
+               libmercury-dev (= 2.0.1~rc1-1),
```
Comment (jolivier23) [inline debian/control:13]: is there no change log for this ?

**Review by jolivier23 — DISMISSED** [review]

File: debian/control (hunk 1 above)
Comment (brianjmurrell) [reply debian/control:13]: Yeah, there should be.

**Review by brianjmurrell — CHANGES_REQUESTED** [review]

File: debian/control (hunk 1 above)
Comment (frostedcmos) [reply debian/control:13]: added new one, though i thought previous entry would be sufficient

**Review by jolivier23 — APPROVED** [review]

**Review by brianjmurrell — COMMENTED** [review]
Packaging changes LGTM.

I re-requested review from me to see if it would clear the -1.  Not sure if it did or not.  My review might need dismissing but I'll leave that to a gatekeeper to do.

File: src/cart/crt_hg.c [hunk 2]
```diff
@@ -1093,27 +1095,15 @@ crt_hg_req_send(struct crt_rpc_priv *rpc_priv)
...
-	 * For network errors, we do not want to return error back to the caller
-	 * of crt_req_send(), but instead we want to invoke completion callback
-	 * manually with 'node unreachable' (DER_UNREACH) error.
-	 *
-	 * HG_NA_ERROR and HG_PROTOCOL_ERROR are both network-level errors
-	 * that can be raised by mercury.
-	 */
-	if (crt_hg_network_error(hg_ret)) {
+	/* For any error to be reported via completion callback */
+	if (hg_ret != HG_SUCCESS) {
```
Comment (gnailzenh) [inline src/cart/crt_hg.c:1099]: do we want to differentiate errors, e.g upper level stack might want to retry for HG_TIMEOUT or HG_AGAIN, but it does not make sense to retry for HG_PERMISSION or HG_PROTOCOL_ERROR

**Review by gnailzenh — APPROVED** [review]

File: src/cart/crt_hg.c (hunk 2 above)
Comment (frostedcmos) [reply src/cart/crt_hg.c:1099]: I ll make a note to add propagation of error into callback to let caller differentiate between errors

**Pull request merged**